.github/workflows/test.yml

```
@@ -0,0 +1,102 @@
+name: Build & Test
+
+on:
+  push:
+    branches: [master]
+  pull_request:
```

```suggestion
  pull_request:
    branches:
      - '*'
```

its just more "explicit"

---

ps: this only makes sense if you want to run those jobs in all the PRs, otherwise I'd remove the `pull_request` altogether